src/OpenTelemetry.Instrumentation.Wcf/TelemetryEndpointBehavior.cs

```diff
@@ -62,8 +66,25 @@ public void Validate(ServiceEndpoint endpoint)
     {
     }
 
+    internal static void ApplyBindingElementToServiceEndpoint(ServiceEndpoint endpoint)
+    {
+        if (endpoint.Binding is CustomBinding customBinding && customBinding.Elements.Find<TelemetryBindingElement>() != null)
```

Comment:
If this runs on the service side too, you should probably also skip if `endpoint.IsSystemEndpoint` is `true`.

Reply:
it won't currently ever run on the service-side, but it doesn't hurt to have it there 👍